src/hooks/users.ts

```diff
@@ -1,52 +1,61 @@
 import { useMemo } from "react";
-import { useFirestoreConnect } from "react-redux-firebase";
+import { isLoaded } from "react-redux-firebase";
 
 import { usersSelector, usersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
 import { useSelector } from "hooks/useSelector";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 import { useVenueId } from "hooks/useVenueId";
+import { useSparkleFirestoreConnect } from "hooks/useSparkleFirestoreConnect";
 
 import { User } from "types/User";
 
 const useConnectUsers = () => {
-  useFirestoreConnect("users");
+  const venueId = useVenueId();
+
+  useSparkleFirestoreConnect([
+    {
+      collection: "users",
+      where: ["enteredVenueIds", "array-contains", venueId],
+      storeAs: "users",
+    },
+  ]);
 };
 
-export const usePartygoers = () => {
+export const usePartygoers = (): readonly WithId<User>[] => {
   useConnectUsers();
 
   const lastSeenThreshold = useUserLastSeenThreshold();
 
-  const users = useSelector(usersSelector);
+  const users = useSelector(usersSelector) ?? [];
 
   return useMemo(
     () => users.filter((user) => user.lastSeenAt > lastSeenThreshold),
     [users, lastSeenThreshold]
   );
 };
 
-export const useUsersById = () => {
+export const useUsersIsLoaded = () => {
```

Comment:
[nit] Follow our `isX` boolean naming pattern.

```suggestion
export const useIsUsersLoaded = () => {
```